### DISCUSSION TITLE
Generator that combines node/element sets?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#21682] [Q&A General] [answered]
Hi all -

Is there a MeshGenerator that takes multiple node sets or element sets as input and combines them into a single, new set?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the RenameBoundary and RenameBlock Generator to combine node / respectively element sets

Guillaume

**Reply by jessecarterMOOSE** [reply]
Does that keep the old sets intact?

**Reply by GiudGiud** [reply]
no.
why do you need to combine them?
we write most objects such that one can pass inputs like 'sideset1 sideset2' and it ll still work

**Reply by jessecarterMOOSE** [reply]
I'm trying to read in Abaqus mesh files, which have this combining feature built-in, so I'm looking for an equivalent in MOOSE. I may be able to use your solution as long as individual node/element sets are not referred to.
